worker-testing-integration/src/test/java/com/hpe/caf/worker/testing/configuration/FileInputWorkerTaskFactoryTest.java

```diff
@@ -0,0 +1,107 @@
+
+import java.io.File;
+import java.io.InputStream;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.nio.file.FileSystemException;
+import java.util.UUID;
+
+public class FileInputWorkerTaskFactoryTest {
+
+    class TestFileInputWorkerTaskFactory extends FileInputWorkerTaskFactory {
+
+        public TestFileInputWorkerTaskFactory(DataStore dataStore, String containerId, String testFilesFolder,
+                                              String testSourcefileBaseFolder,
+                                              String overrideReference) throws Exception {
+            super(dataStore, containerId, testFilesFolder, testSourcefileBaseFolder, overrideReference);
+        }
+
+        @Override
+        protected Object createTask(TestItem testItem, ReferencedData sourceData) {
+            return null;
+        }
+
+        @Override
+        public String getWorkerName() {
+            return null;
+        }
+
+        @Override
+        public int getApiVersion() {
+            return 0;
+        }
+    }
+
+    public FileInputWorkerTaskFactoryTest() throws Exception {
+    }
+
+    @Test
+    public void testFileInputWorkerTaskFactoryCreateTaskDoesNotThrowFileSystemException () throws Exception {
+
+        // Mock DataStore
+        DataStore mockDataStore = Mockito.mock(DataStore.class);
+        Mockito.when(mockDataStore.store(Mockito.any(InputStream.class), Mockito.any(String.class)))
+                .thenReturn("mockRefId");
+
+        String containerId = "mockContainerId";
+        String testFilesFolder = "";
+        String testSourcefileBaseFolder = "";
+        TestFileInputWorkerTaskFactory testFileInputWorkerTaskFactory = new TestFileInputWorkerTaskFactory(
+                mockDataStore, containerId, testFilesFolder, testSourcefileBaseFolder, null);
+
+        OperatingSystemMXBean os = ManagementFactory.getOperatingSystemMXBean();
+        if(os instanceof UnixOperatingSystemMXBean){
+            int numberOfTimesToRun = (int) ((UnixOperatingSystemMXBean) os).getMaxFileDescriptorCount() + 1;
+            System.out.println("Number of times to run create task: " + numberOfTimesToRun);
+            for (int i = 0; i < numberOfTimesToRun; i++) {
+                FileTestInputData fileTestInputData = new FileTestInputData();
+                ClassLoader classLoader = getClass().getClassLoader();
+                String mockFilePath = new File(classLoader.getResource("mockFile.txt").getFile()).getAbsolutePath();
+                fileTestInputData.setInputFile(mockFilePath);
+                fileTestInputData.setUseDataStore(true);
+
+                TestItem testItem = new TestItem("mockFile.txt", fileTestInputData, null);
+                testItem.setInputIdentifier(UUID.randomUUID().toString());
+                try {
+                    testFileInputWorkerTaskFactory.createTask(testItem);
+                } catch (Exception exception) {
```

Comment:
Wouldn't it be better to just not catch any exceptions here? I assume the test doesn't throw exceptions if it succeeds?
Or maybe just catch the FileSystemException to output some meaningful error message and then re-throw?